pkg/kubelet/util/cache/version_cache.go

```diff
@@ -17,14 +17,12 @@ limitations under the License.
 package cache
 
 import (
-	"fmt"
 	"sync"
 	"time"
 
 	"github.com/golang/glog"
 
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
-	"k8s.io/kubernetes/pkg/util/wait"
 )
 
 type VersionCache struct {
```

Comment:
Because the cache is placed in kubelet/util, I think something like an ObjectCache makes more sense to me. :)

You can pass a `updater` function and set expiration when creating the cache, and just let the cache store an object.